### PR TITLE
Docs: mark up dbm.open() with param list

### PR DESCRIPTION
**Pull request by erlend-aasland**
Also consolidate following paragraphs regarding database objects.

----
📚 Documentation preview 📚: https://cpython-previews--114601.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/dbm.rst [hunk 1]
```diff
@@ -58,41 +58,33 @@ the Oracle Berkeley DB.
...
+   :param int mode:
+      The Unix file access mode of the file (default: octal ``0o666``),
+      used only when the database has to be created.
 
    .. versionchanged:: 3.11
       *file* accepts a :term:`path-like object`.
 
-
-The object returned by :func:`open` supports the same basic functionality as a
+The object returned by :func:`~dbm.open` supports the same basic functionality as a
```
Comment (erlend-aasland) [inline Doc/library/dbm.rst:84]: Explicitly use `~dbm.open`, so there is no doubt we refer to dbm.open, and not the builtin.

File: Doc/library/dbm.rst [hunk 2]
```diff
@@ -101,9 +93,17 @@ being stored.
...
+   :meth:`!get` and :meth:`!setdefault` methods are now available for all
+   :mod:`dbm` backends.
+
 .. versionchanged:: 3.4
    Added native support for the context management protocol to the objects
-   returned by :func:`.open`.
+   returned by :func:`~dbm.open`.
+
+.. versionchanged:: 3.8
+   Deleting a key from a read-only database raises database module specific error
```
Comment (erlend-aasland) [inline Doc/library/dbm.rst]:
```suggestion
   Deleting a key from a read-only database raises a database module specific exception
```

File: Doc/library/dbm.rst [hunk 3]
```diff
@@ -58,41 +58,33 @@ the Oracle Berkeley DB.
...
 
-      ``'r'`` (default), |flag_r|
-      ``'w'``, |flag_w|
-      ``'c'``, |flag_c|
-      ``'n'``, |flag_n|
+   :param str flag:
+      * ``'r'`` (default), |flag_r|
+      * ``'w'``, |flag_w|
+      * ``'c'``, |flag_c|
+      * ``'n'``, |flag_n|
```
Comment (erlend-aasland) [inline Doc/library/dbm.rst:75]: A list of flags renders nicer and more compact than a table, when embedded in the param list.

**Review by ezio-melotti — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review, Ezio!

**Pull request merged**